astropy/time/core.py

```diff
@@ -686,8 +686,8 @@ def shape(self, shape):
 
     def _shaped_like_input(self, value):
         out = value
-        if not self._time.jd1.shape and not np.ma.is_masked(value):
-            out = value.item()
+        # if not self._time.jd1.shape and not np.ma.is_masked(value):
+        #     out = value.item()
```

Comment:
I had to comment this out because with it, the Structured array was converted into an ordinary tuple.

I do not know how to handle this.

Reply:
I've had this problem elsewhere as well; instead do:
```
out = value[()] if value.dtype.names else value.item()
```
(The `[()]` turns it into an array scalar, which is what we want here.)

Reply:
Doing that did not work.